example/fine_advection/stokes.hpp

```diff
@@ -0,0 +1,150 @@
+//========================================================================================
+// (C) (or copyright) 2020-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#ifndef EXAMPLE_FINE_ADVECTION_STOKES_HPP_
+#define EXAMPLE_FINE_ADVECTION_STOKES_HPP_
+
+#include <memory>
+#include <vector>
+
+#include <parthenon/driver.hpp>
+#include <parthenon/package.hpp>
+
+namespace advection_example {
+using namespace parthenon::driver::prelude;
+
+template <class pack_desc_t>
+TaskStatus WeightedSumData(parthenon::CellLevel cl, parthenon::TopologicalElement te,
+                           pack_desc_t &pd, MeshData<Real> *in1, MeshData<Real> *in2,
+                           Real w1, Real w2, MeshData<Real> *out) {
+  auto pack1 = pd.GetPack(in1);
+  auto pack2 = pd.GetPack(in2);
+  auto pack_out = pd.GetPack(out);
+
+  IndexRange ib = in1->GetBoundsI(cl, IndexDomain::entire, te);
+  IndexRange jb = in1->GetBoundsJ(cl, IndexDomain::entire, te);
+  IndexRange kb = in1->GetBoundsK(cl, IndexDomain::entire, te);
+
+  parthenon::par_for(
+      PARTHENON_AUTO_LABEL, 0, pack1.GetNBlocks() - 1, pack1.GetLowerBoundHost(0),
+      pack1.GetUpperBoundHost(0), // This is safe for dense vars
+      kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int l, const int k, const int j, const int i) {
+        pack_out(b, l, k, j, i) = w1 * pack1(b, l, k, j, i) + w2 * pack2(b, l, k, j, i);
+      });
+  return TaskStatus::complete;
+}
```

Comment:
Should this be in `interface/update`?

Reply:
Eventually, but I think for now we should leave it in the example. I was trying to write a parallel set of functionality to `update` here that works with more general fields.